### PR TITLE
dal2: Polish CallbackReader implement

### PR DESCRIPTION
**Pull request by Xuanwo**
Signed-off-by: Xuanwo <github@xuanwo.io>

I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

This PR polish the implementations for `CallbackReader`.

## Changelog

- Improvement

## Test Plan

Unit Tests

Stateless Tests

### CODE REVIEW

File: common/dal2/src/ops/io.rs [hunk 1]
```diff
@@ -102,6 +108,7 @@ impl futures::AsyncRead for CallbackReader {
 
 #[cfg(test)]
 mod test {
```
Comment (PsiACE) [inline common/dal2/src/ops/io.rs:110]: we can add a src/tests for inner tests, or move it to tests/ut

Comment (PsiACE) [reply common/dal2/src/ops/io.rs:110]: https://github.com/datafuselabs/databend/discussions/3473 for testing style

**Comment by Xuanwo** [comment]
Addressed, PTAL.

File: common/dal2/src/ops/io.rs [hunk 2]
```diff
@@ -99,24 +117,3 @@ impl futures::AsyncRead for CallbackReader {
         r
     }
 }
-
```
Comment (BohuTANG) [inline common/dal2/src/ops/io.rs:102]: 💯

**Review by PsiACE — APPROVED** [review]
LGTM

**Pull request merged**